frontend/packages/metal3-plugin/src/components/baremetal-hosts/host-menu-actions.tsx

```diff
@@ -71,10 +77,36 @@ export const PowerOn = (kindObj: K8sKind, host: BareMetalHostKind): KebabOption
   };
 };
 
+export const Deprovision = (
+  kindObj: K8sKind,
+  host: BareMetalHostKind,
+  { machine, machineSet }: ActionArgs,
+): KebabOption => {
+  const title = 'Deprovision';
+  return {
+    hidden:
+      !machine ||
+      !!getAnnotations(machine, {})[DELETE_MACHINE_ANNOTATION] ||
```

Comment:
That would be a bug on the machineSet side. I think it is much safer to rely on the annotation because the status can take a while to get updated and the user might remove two machines by accident (will randomly kill next one).

I added a safety fallback to the request so the annotation doesn't get stuck when the machine set patch fails